src/ar/hentaislayer/src/eu/kanade/tachiyomi/extension/ar/hentaislayer/HentaiSlayer.kt

```diff
@@ -0,0 +1,245 @@
+            .map { it.text() }
+            .let { listOf(genre) + it }
+            .joinToString()
+    }
+
+    override fun popularMangaNextPageSelector() = "ul.pagination > li:last-child:not(.pagination-disabled)"
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/latest-${getLatestTypes()}?page=$page", headers)
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/manga?title=$query".toHttpUrl().newBuilder()
+        filters.forEach { filter ->
+            when (filter) {
+                is TypeFilter -> url.addQueryParameter("type", filter.toUriPart())
+                is StatusFilter -> url.addQueryParameter("status", filter.toUriPart())
+                is GenresFilter ->
+                    filter.state
+                        .filter { it.state }
+                        .forEach { url.addQueryParameter("genre[]", it.uriPart) }
+                else -> {}
+            }
+        }
+
+        url.addQueryParameter("page", page.toString())
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        document.selectFirst("main section")?.run {
+            selectFirst("img#manga-cover")?.run {
+                thumbnail_url = imgAttr()
+                title = attr("alt")
+            }
+            selectFirst("section > div:nth-child(1) > div:nth-child(1) > div:nth-child(2) > div:nth-child(2)")?.run {
+                status = parseStatus(select("a[href*='?status=']").text())
+                genre = select("a[href*='?type=']").text()
+                author = select("p:has(span:contains(المؤلف)) span:nth-child(2)").text()
+                artist = select("p:has(span:contains(الرسام)) span:nth-child(2)").text()
+            }
+            selectFirst("section > div:nth-child(1) > div:nth-child(2)")?.run {
+                select("h1").text().takeIf { it.isNotEmpty() }?.let {
+                    title = it
+                }
+                genre = select("a[href*='?genre=']")
+                    .map { it.text() }
+                    .let {
+                        listOf(genre) + it
```

Comment:
consider saving the genre from above in a variable and building the genre string once.